### ISSUE TITLE
mzML bombs

### ISSUE THREAD

**Issue by hroest** [OpenMS/OpenMS#1636]
Due to the way that OpenMS stores the DataProcessing parameters, it is possible to observe real-world and artificially generated "mzML bombs", i.e. a small mzML file that consumes large amounts of memory. This usually happens in situations where many scans are present and the file has been processed with multiple tools that each have a large number of parameters. 

Generating such a file is quite easy, see the following Python script:
[generate_mzML_bomb.txt](https://github.com/OpenMS/OpenMS/files/7112/generate_mzML_bomb.txt)

Running the 1.11 OpenMS FileConverter on this datafile uses a huge amount of memory:

```
$ /usr/bin/time FileConverter -in test.mzML.bz2 -out /tmp/t.mzML
Progress of 'loading mzML file':
-- done [took 8.55 s (CPU), 9.69 s (Wall)] --
Progress of 'storing mzML file':
-- done [took 4.37 s (CPU), 4.44 s (Wall)] --
FileConverter took 25.21 s (wall), 23.90 s (CPU), 0.00 s (system), 23.90 s (user).
23.92user 1.38system 0:25.44elapsed 99%CPU (0avgtext+0avgdata 11846864maxresident)k
0inputs+7056outputs (0major+741578minor)pagefaults 0swaps
```
-> using 2892.3 MB of memory to read a 25 KB file!

This can be solved when using a FileConverter as provided by PR #695 :  

```
$ /usr/bin/time FileConverter -in test.mzML.bz2 -out /tmp/t.mzML
Progress of 'loading spectra list':
-- done [took 0.40 s (CPU), 0.41 s (Wall)] --
Progress of 'storing mzML file':
-- done [took 1.20 s (CPU), 1.21 s (Wall)] --
FileConverter took 2.03 s (wall), 1.99 s (CPU), 0.00 s (system), 1.99 s (user).
2.02user 0.03system 0:02.06elapsed 99%CPU (0avgtext+0avgdata 162032maxresident)k
0inputs+7048outputs (0major+11133minor)pagefaults 0swaps
```

This approach only uses 39.3 MB of memory which is substantially more reasonable. (It is still not optimal, however also reading an empty mzML file uses about 18.8 MB of memory).

**Comment by timosachsenberg** [comment]
Interesting. should we consider this resolved with PR #695 ?

**Comment by hroest** [comment]
yes, this should be resolved now
